### PR TITLE
Gen 4: Ensure Brick Break breaks screens of immune targets

### PR DESCRIPTION
**Pull request by vivapuffs**
https://github.com/smogon/pokemon-showdown/projects/3#card-61615516
This is my first time contributing to an open source project, please let me know if I've made any mistakes

### CODE REVIEW

File: data/mods/gen4/moves.ts [hunk 1]
```diff
@@ -178,6 +178,18 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		inherit: true,
 		recoil: [1, 3],
 	},
+	brickbreak: {
+		inherit: true,
+		ignoreImmunity: true,
+		onTryHit(pokemon, target){
+			pokemon.side.removeSideCondition('reflect');
+			pokemon.side.removeSideCondition('lightscreen');
+			if (pokemon.hasType('Ghost')){
```
Comment (KrisXV) [inline data/mods/gen4/moves.ts:187]:
```suggestion
			if (pokemon.hasType('Ghost')) {
```

**Comment by MathyFurret** [comment]
Hi! There's a unit test written for this here: https://github.com/smogon/pokemon-showdown/blob/master/test/sim/moves/brickbreak.js#L41 Could you delete the ".skip" so it runs?

**Comment by vivapuffs** [comment]
> Hi! There's a unit test written for this here: https://github.com/smogon/pokemon-showdown/blob/master/test/sim/moves/brickbreak.js#L41 Could you delete the ".skip" so it runs?

oh whoops should be done now

**Comment by urkerab** [comment]
What if the user's ability is Scrappy?

**Comment by vivapuffs** [comment]
good catch, is the fix i put in my latest commit appropriate?

**Comment by urkerab** [comment]
You should use `hasAbility` to check for an ability, as that takes Gastro Acid into account. You should also use `runImmunity` to check for an immunity, as that allows Foresight to negate the immunity.

**Comment by vivapuffs** [comment]
it appears to work correctly with just runImmunity (see latest commit,) tested against ghost mon, ghost mon under foresight, and scrappy mon vs. ghost mon

**Pull request merged**